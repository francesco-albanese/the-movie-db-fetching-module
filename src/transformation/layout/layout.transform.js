import { isEmpty } from 'lodash-es'
import { getSections } from '#utils'

export const transformLayoutResponse = data => {
  
  if (isEmpty(data)) {
    return {}
  }

  const { items, includes } = data

  return items.map(template => {

    const sections = getSections(template, includes.Entry)

    return {
      sections
    }
  })

}