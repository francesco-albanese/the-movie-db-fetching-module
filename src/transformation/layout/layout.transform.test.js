import { transformLayoutResponse } from './layout.transform'
import * as Fixtures from './__fixtures__/layout.transform.fixtures'

describe('transformLayoutResponse', () => {
  it('returns an empty array if data is missing', () => {
    expect(transformLayoutResponse()).toEqual([])
  })

  it('returns transformed layout', () => {
    const result = transformLayoutResponse(Fixtures.data)
    expect(result).toMatchSnapshot()
  })
})