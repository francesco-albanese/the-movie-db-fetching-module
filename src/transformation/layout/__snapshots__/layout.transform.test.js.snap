// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformLayoutResponse returns transformed layout 1`] = `
Array [
  Object {
    "sections": Object {
      "MainMenu": Array [
        Object {
          "Home": Object {
            "exact": Object {
              "en-GB": false,
            },
            "icon": Object {
              "en-GB": "Home",
            },
            "label": Object {
              "en-GB": "Home",
              "it": "Home",
            },
            "name": Object {
              "en-GB": "Home",
            },
            "path": Object {
              "en-GB": "/en",
              "it": "/it",
            },
          },
        },
        Object {
          "NowPlaying": Object {
            "exact": Object {
              "en-GB": false,
            },
            "icon": Object {
              "en-GB": "PlayArrow",
            },
            "label": Object {
              "en-GB": "Now Playing",
              "it": "Adesso In Onda",
            },
            "name": Object {
              "en-GB": "NowPlaying",
            },
            "path": Object {
              "en-GB": "/now-playing",
              "it": "/adesso-in-onda",
            },
          },
        },
        Object {
          "Popular": Object {
            "exact": Object {
              "en-GB": false,
            },
            "icon": Object {
              "en-GB": "LocalMovies",
            },
            "label": Object {
              "en-GB": "Popular Movies",
              "it": "Film Popolari",
            },
            "name": Object {
              "en-GB": "Popular",
            },
            "path": Object {
              "en-GB": "/popular",
              "it": "/popolari",
            },
          },
        },
        Object {
          "TopRated": Object {
            "exact": Object {
              "en-GB": false,
            },
            "icon": Object {
              "en-GB": "StarRate",
            },
            "label": Object {
              "en-GB": "Top Rated",
              "it": "Premiatissimi",
            },
            "name": Object {
              "en-GB": "TopRated",
            },
            "path": Object {
              "en-GB": "/top-rated",
              "it": "/premiatissimi",
            },
          },
        },
        Object {
          "Upcoming": Object {
            "exact": Object {
              "en-GB": false,
            },
            "icon": Object {
              "en-GB": "RecentActors",
            },
            "label": Object {
              "en-GB": "Upcoming",
              "it": "In Arrivo",
            },
            "name": Object {
              "en-GB": "Upcoming",
            },
            "path": Object {
              "en-GB": "/upcoming",
              "it": "/in-arrivo",
            },
          },
        },
      ],
    },
  },
]
`;
